spec/formatting.md

```diff
@@ -1,24 +1,124 @@
+or even use them in their internal processing,
+as long as the final _formatting_ result is made available to users
+and the observable behavior of the formatter matches that described here.
+
+## Formatting Context
+
+A message's **_formatting context_** represents the data and procedures that are required
+for the message's _resolution_, _pattern selection_ and _formatting_.
+
+At a minimum, it includes:
+
+- Information on the current **_locale_**,
+  potentially including a fallback chain of locales.
+  This will be passed on to formatting functions.
+
+- Information on the base directionality of the message and its _text_ tokens.
+  This will be used by strategies for bidirectional isolation.
+
+- A mapping of string identifiers to values,
+  defining variable values that may be used during _variable resolution_.
+  This is often determined by a user-provided argument of a formatting function call.
+
+- The _function registry_,
+  providing the implementations of the functions referred to by message _functions_.
+
+- Optionally, a fallback string to use for the message
+  if it contains any Syntax or Data Model errors.
+
+Implementations MAY include additional fields in their _formatting context_.
+
+## Resolution
+
+### Expression Resolution
+
+_Expressions_ are used in _declarations_, _selectors_, and _patterns_.
+
+In a _declaration_, the resolved value of the _expression_ is assigned to a _variable_,
+which may then be used in other _expressions_.
+As such a _variable_ MAY then be referenced in different ways,
+implementations SHOULD NOT immediately fully format the value for output.
+
+In _selectors_, the resolved value of an _expression_ is used for _pattern selection_.
+
+In a _pattern_, the resolved value of an _expression_ is used in its _formatting_.
+
+Depending on the presence or absence of an _operand_
+and a _function_ or _reserved_ _annotation_,
+one of the following is used to resolve the value of the _expression_:
+
+- If the _expression_ contains no _annotation_,
+  its resolved value is determined by _literal resolution_ or _variable resolution_,
+  depending on the shape of the _operand_.
+- Else, if the _expression_ has a _function_ _annotation_,
+  its resolved value is defined by _function resolution_.
+- Else, the _expression_ has a _reserved_ _annotation_,
+  and a fallback value is used as its value.
+
+### Literal Resolution
+
+The resolved value of a _text_ or a _literal_ is
+the character sequence of the _text_ or _literal_
+after any character escape has been converted to the escaped character.
+
+When a _literal_ is used as an _operand_
+or on the right-hand side of an _option_,
+the formatting function MUST treat its resolved value the same
+independently of whether the value was originally _quoted_ or _unquoted_.
+
+> For example,
+> the _option_ `foo=42` and the _option_ `foo=|42|` are treated as identical.
+
+The resolution of a _text_ or _literal_ token MUST always succeed.
```

Comment:
I don't think the term "token" has been defined yet.

Reply:
It's from here: https://github.com/unicode-org/message-format-wg/blob/main/spec/syntax.md#tokens

Reply:
Can that be cited in this document?

Reply:
I believe it effectively already is, as we've established in `spec/README.md` that e.g. the italic "_text_" is a reference to the bold-italic "**_text_**" definition that's a subsection of the "Tokens" section of `syntax.md`.

I don't know how exactly we'll do the final rendering of this, but I would presume that we would at that stage linkify these terms accordingly.